assets/js/script.js

```diff
@@ -0,0 +1,164 @@
+const forecastContainer = $("#forecast-container");
+
+const constructSearchHistory = function (array) {
+  array.forEach((element) => {
+    historyButton = `<li><button id="search-btn">${element}</button></li>`;
+    const searchHist = $("#search-history");
+    searchHist.append(historyButton);
+  });
+};
+
+const renderSearchHistory = function () {
+  searchContainer.empty();
+  const array = getFromLocalStorage();
+  constructSearchHistory(array);
+};
+
+const constructTodaysWeather = function (data) {
+  const date = moment.unix(data.dt).format("DD-MM-YYYY");
+
+  return `<div>
+    <h2>${data.name} : ${date}
+        <img src="http://openweathermap.org/img/wn/${data.weather[0].icon}.png">
+    </h2>
+    <ul class="list-group-flush">
+        <li class="custom">Temperature: ${data.main.temp} <sup>O</sup>C</li>
+        <li class="custom">Humidity: ${data.main.humidity} %</li>
+        <li class="custom">Wind Speed: ${data.wind.speed} m/s</li>
+        <li class="custom" >UV Index: <span id="uvi"></span></li>
+      </ul>
+  </div>`;
+};
+
+const renderTodaysWeather = function (data) {
+  currentWeather.empty();
+  const topBlock = constructTodaysWeather(data);
+  currentWeather.append(topBlock);
+};
+
+const constructForecast = function (dailyArray) {
+  dailyArray.forEach((element) => {
+    const date = moment.unix(element.dt).format("DD-MM-YYYY");
+    forecastBlock = `<div class="card col-2 padstyle" style="width: 18rem;">
+    <div class="card-body">
+      <h5 class="card-title">${date}<img src="http://openweathermap.org/img/wn/${element.weather[0].icon}.png" alt="..."></h5>
+    </div>
+    <ul class="list-group list-group-flush">
+      <li class="list-group-item">Temperature: ${element.temp.day} <sup>O</sup>C</li>
+      <li class="list-group-item">Humidity: ${element.humidity} %</li>
+      <li class="list-group-item">Wind speed: ${element.wind_speed} m/s</li>
+    </ul>
+</div>`;
+    forecastContainer.append(forecastBlock);
+  });
+};
+
+const renderForecast = function (dailyArray) {
+  forecastContainer.empty();
+  constructForecast(dailyArray);
+};
+
+const getFromLocalStorage = function () {
+  const searchHistory = JSON.parse(localStorage.getItem("history")) || [];
+  return searchHistory;
+};
+
+const handleResponse = function (response) {
+  return response.json();
+};
+
+const showError = function () {
+  const error = document.createElement("p");
+  error.textContent = "please enter a valid city";
+  $("#error-response").append(error);
+};
+
+const renderUviSpan = function (uviData) {
+  const placeHolder = $("#uvi");
+  placeHolder.text(uviData);
+  if (uviData < 3) {
+    placeHolder.addClass("green");
+  } else if (uviData >= 3 && uviData < 8) {
+    placeHolder.addClass("orange");
+  } else {
+    placeHolder.addClass("red");
+  }
+};
+
+const renderPage = function (city) {
+  const myUrl =
+    `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&APPID=` +
+    myAPIKey;
```

Comment:
We should use a template string here rather than the string concatenation you have used. Eg. 

`const myUrl =
    `https://api.openweathermap.org/data/2.5/weather?q=${city}&units=metric&APPID=${myAPIKey}`
   `

Reply:
oh yeah, I'm not sure what I was thinking